lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -1784,13 +1785,42 @@ void StackAllocationPromoter::promoteAllocationToPhi() {
 }
 
 void StackAllocationPromoter::run() {
+  auto *function = asi->getFunction();
+
   // Reduce the number of load/stores in the function to minimum.
   // After this phase we are left with up to one load and store
   // per block and the last store is recorded.
   pruneAllocStackUsage();
 
   // Replace AllocStacks with Phi-nodes.
   promoteAllocationToPhi();
+
+  // Make sure that all of the allocations were promoted into registers.
+  assert(isWriteOnlyAllocation(asi) && "Non-write uses left behind");
+
+  SmallVector<SILValue> valuesToComplete;
+
+  // Enum types may have incomplete lifetimes in address form, when promoted to
+  // value form after mem2reg, they will end up with incomplete ossa lifetimes.
+  // Use the lifetime completion utility to complete such lifetimes.
+  // First, collect the stored values to complete.
+  if (asi->getType().isOrHasEnum()) {
+    for (auto it : initializationPoints) {
+      auto *si = it.second;
+      auto src = si->getOperand(0);
```

Comment:
Should `StackAllocationPromoter::getLiveOutValues(_, it.first)` be used here to find the lexical borrows or moves that may have been introduced?

Reply:
Lifetime completion works on inner borrows as well. But looks like it doesn't for inner moves. Also lifetime completion of lexical values (move_value [lexical]) is only possible at the unreachable blocks, what we really need is lifetime completion at the boundary in this enum case in mem2reg. Looks like some additional work is needed here..

Reply:
Going to wait for @atrick's comment here about the lifetime completion utility's use in this case. 

@nate-chandler  Until then, I have disabled this for lexical stack locs.

Reply:
@meg-gupta @nate-chandler For the record...
Lifetime completion only handles inner scopes that might affect the lifetime of the outer scope. It doesn't make sense for lifetime completion to look past any consumes.
This will, however, change when we fully support a `mark_dependence` where an owned value depends on a borrow scope. To handle that, we'll need to look through all potentially forwarding consumes. The final destroy will need to be within the borrow scope. That change only affects cases where such a mark_dependence exists